pyfr/plugins/writer.py

```diff
@@ -58,6 +65,18 @@ def __call__(self, intg):
         else:
             metadata = None
 
+        # Fetch data from other plugins and add it to metadata with ad-hoc keys
+        for csh in intg.completed_step_handlers:
+            try:
+                opdata = csh.get_data()
+
+                if rank == root:
+                    prefix = self._get_plugin_data_prefix(csh.name, csh.suffix)
```

Comment:
I'd get rid of this method as it is only called in one place.

Reply:
I'd keep it as we might need it when restoring data

Reply:
I'm not sure how it could be used, as the method is attached to the writer rather than the base plugin/integrator?